osbenchmark/workload_generator/helpers.py

```diff
@@ -125,21 +135,29 @@ def process_queries(self):
 
         return processed_queries
 
-def process_indices(indices, document_frequency, indices_docs_mapping):
+def process_indices(indices, sample_frequency_mapping, indices_docs_mapping):
     processed_indices = []
     for index_name in indices:
         try:
-            #Setting number_of_docs_for_index to None means OSB will grab all docs available in index
+            # Setting number_of_docs_for_index to None means OSB will grab all docs available in index
             number_of_docs_for_index = None
             if indices_docs_mapping and index_name in indices_docs_mapping:
                 number_of_docs_for_index = int(indices_docs_mapping[index_name])
                 if number_of_docs_for_index <= 0:
                     raise exceptions.SystemSetupError(
                         "Values specified with --number-of-docs must be greater than 0")
 
+            # Do this if sample frequency is specified
+            sample_frequency_for_index = None
+            if sample_frequency_mapping and index_name in sample_frequency_mapping:
+                sample_frequency_for_index = int(sample_frequency_mapping[index_name])
+                if sample_frequency_for_index <= 1:
```

Comment:
As mentioned above, the value `1` could be a degenerate case where every doc is extracted.